### DISCUSSION TITLE
where is agenix private key being referenced?

### DISCUSSION THREAD

**Discussion by ryanmsnyder** [dustinlyons/nixos-config#64] [Q&A] [answered]
Hi Dustin,

I've been using your non-secrets config but am now trying to implement `agenix` into it by referencing your secrets template modules. The README states:

---
<!DOCTYPE html><p dir="auto" style="box-sizing: border-box; margin-top: 0px; margin-bottom: 16px; caret-color: rgb(230, 237, 243); color: rgb(230, 237, 243); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, &quot;Noto Sans&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 16px; font-style: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none;">Before generating your first build, these keys must exist in your<span class="Apple-converted-space"> </span><code style="box-sizing: border-box; font-family: var(--fontStack-monospace, ui-monospace, SFMono-Regular, SF Mono, Menlo, Consolas, Liberation Mono, monospace); font-size: 13.6px; padding: 0.2em 0.4em; margin: 0px; white-space: break-spaces; background-color: var(--bgColor-neutral-muted, var(--color-neutral-muted)); border-radius: 6px;">~/.ssh</code><span class="Apple-converted-space"> </span>directory. Don't worry, I provide a few commands to help you.</p>
Key Name | Platform | Description
-- | -- | --
id_ed25519 | macOS / NixOS | Used to download secrets from Github.
id_ed25519_agenix | macOS / NixOS | Used to encrypt and decrypt secrets.

<br class="Apple-interchange-newline">

---
However, I noticed in `secrets.nix`, the `identityFile` doesn't reference `id_ed25519_agenix`. It references `id_ed25519`. I couldn't figure out where `id_ed25519_agenix` is being used. Am I misunderstanding something? Thanks again for your config and help.

**Comment by dustinlyons** [comment, marked as answer]
Hi, this is somewhat confusing and I should eventually make it more clear.

The first key `id_ed25519` is used just to download your `nix-secrets` repo on initial bootstrap, assuming it's private.

The second key, `id_ed25519_agenix` is copied over to your system and *renamed* to `id_ed25519` during this process. It's then referenced by `agenix` to do encryption and decryption.
